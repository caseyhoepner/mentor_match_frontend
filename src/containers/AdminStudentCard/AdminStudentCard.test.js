import React from 'react';
import { shallow } from 'enzyme';
import { AdminStudentCard } from './';

describe ('AdminStudentCard', () => {
  let wrapper;
  let mockStudent = { 
      name: 'Casey',
      identity_marker: [ 'female-identifying' ],
      stack: 'Front-End',
      matched: 'true'
    }

  beforeEach(() => {
    wrapper = shallow(<AdminStudentCard student={mockStudent} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})