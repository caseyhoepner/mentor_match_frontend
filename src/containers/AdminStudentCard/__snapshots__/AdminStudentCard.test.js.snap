// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminStudentCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminStudentCard
    student={
      Object {
        "identity_marker": Array [
          "female-identifying",
        ],
        "matched": "true",
        "name": "Casey",
        "stack": "Front-End",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="asc-name"
        >
          Casey
        </p>,
        <p
          className="asc-stack"
        >
          Front-End
        </p>,
        <div
          className="asc-prefs"
        >
          <img
            alt="female-identifying preference indicator"
            className="asc-pref-icon"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "female-identifying.svg",
              }
            }
            title="female-identifying"
          />
        </div>,
        "        ",
        <div
          className="asc-matched"
        >
          <img
            alt="matched indicator"
            className="asc-matched-icon"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "matched-true.svg",
              }
            }
          />
        </div>,
      ],
      "className": "asc-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Casey",
          "className": "asc-name",
        },
        "ref": null,
        "rendered": "Casey",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Front-End",
          "className": "asc-stack",
        },
        "ref": null,
        "rendered": "Front-End",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="female-identifying preference indicator"
              className="asc-pref-icon"
              src={
                Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "female-identifying.svg",
                }
              }
              title="female-identifying"
            />,
          ],
          "className": "asc-prefs",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "female-identifying",
            "nodeType": "host",
            "props": Object {
              "alt": "female-identifying preference indicator",
              "className": "asc-pref-icon",
              "src": Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "female-identifying.svg",
              },
              "title": "female-identifying",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      "        ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="matched indicator"
            className="asc-matched-icon"
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "matched-true.svg",
              }
            }
          />,
          "className": "asc-matched",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "matched indicator",
            "className": "asc-matched-icon",
            "src": Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "matched-true.svg",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="asc-name"
          >
            Casey
          </p>,
          <p
            className="asc-stack"
          >
            Front-End
          </p>,
          <div
            className="asc-prefs"
          >
            <img
              alt="female-identifying preference indicator"
              className="asc-pref-icon"
              src={
                Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "female-identifying.svg",
                }
              }
              title="female-identifying"
            />
          </div>,
          "        ",
          <div
            className="asc-matched"
          >
            <img
              alt="matched indicator"
              className="asc-matched-icon"
              src={
                Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "matched-true.svg",
                }
              }
            />
          </div>,
        ],
        "className": "asc-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Casey",
            "className": "asc-name",
          },
          "ref": null,
          "rendered": "Casey",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Front-End",
            "className": "asc-stack",
          },
          "ref": null,
          "rendered": "Front-End",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="female-identifying preference indicator"
                className="asc-pref-icon"
                src={
                  Object {
                    "ReactComponent": [Function],
                    "__esModule": true,
                    "default": "female-identifying.svg",
                  }
                }
                title="female-identifying"
              />,
            ],
            "className": "asc-prefs",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "female-identifying",
              "nodeType": "host",
              "props": Object {
                "alt": "female-identifying preference indicator",
                "className": "asc-pref-icon",
                "src": Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "female-identifying.svg",
                },
                "title": "female-identifying",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        "        ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="matched indicator"
              className="asc-matched-icon"
              src={
                Object {
                  "ReactComponent": [Function],
                  "__esModule": true,
                  "default": "matched-true.svg",
                }
              }
            />,
            "className": "asc-matched",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "matched indicator",
              "className": "asc-matched-icon",
              "src": Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "matched-true.svg",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
